File: ABC107/c.cpp
#include <bits/stdc++.h>
using namespace std;
#define REP(i, n) for(int i = 0; i < n; i++)

int main(){
    cin.tie(0);
    ios::sync_with_stdio(false);

    int n, k;
    cin >> n >> k;
    vector<int> A(n);
    REP(i, n) cin >> A[i];

    int ans = 1e9;
    for(int i = 0; i < n - k + 1; i++){
        int cost = 0, tmp;
        for(int j = i; j < i + k; j++){
            if(j == i);
            else cost += A[j] - tmp;
            tmp = A[j];
        }
        if(0 < A[i] or A[i + k - 1] < 0){
            cost += min(abs(A[i]), abs(A[i + k - i]));
        }
        cout << i << " " << cost<< endl;
        ans = min(ans, cost);
    }
    cout << ans << endl;

    return 0;
}

File: ABC107/c.py
def main():
    n, k = map(int, input().split())
    A = list(map(int, input().split()))
    
    ans = 0
    cost = 0
    b = 0
    f = False
    for i in range(n - k + 1):
        if i == 0:
            for j in range(k):
                if j != 0:
                    cost += A[j] - A[j - 1]
            if A[i] <= 0 <= A[i + k - 1]:
                pass
            else:
                f = True
                b =  min(abs(A[i]), abs(A[k - 1]))
                cost += b
            ans = cost
        else:
            if f == True:
                cost -= b
                f = False
            cost -= A[i] - A[i - 1]
            cost += A[i + k - 1] - A[i + k - 2]
            if A[i] <= 0 <= A[i + k - 1]:
                pass
            else:
                f = True
                b = min(abs(A[i]), abs(A[i + k - 1]))
                cost += b
            print(ans, cost)
            ans= min(ans, cost)
   
    print(ans)

if __name__ == "__main__":
    main()